airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_manifest_only/manifest_component_transformer.py

```diff
@@ -121,7 +184,9 @@ def propagate_types_and_parameters(
         # Parameters should be applied to the current component fields with the existing field taking precedence over parameters if
         # both exist
         for parameter_key, parameter_value in current_parameters.items():
-            propagated_component[parameter_key] = propagated_component.get(parameter_key) or parameter_value
+            if parameter_key in valid_fields:
```

Comment:
By conditionally propagating the parameter only if it is an expected field returned from the model, we avoid adding unexpected fields that break the builder's UI mode.